NotifyMe/services/service.py

```diff
@@ -0,0 +1,202 @@
+            return start_date + timedelta(days=plans_duration["BASIC"])
+        elif subscription_plan.subscription_plan == plans["REGULAR_PLAN"]:
+            return start_date + timedelta(days=plans_duration["REGULAR"])
+        elif subscription_plan.subscription_plan == plans["STANDARD_PLAN"]:
+            return start_date + timedelta(days=plans_duration["STANDARD"])
+        elif subscription_plan.subscription_plan == plans["PREMIUM_PLAN"]:
+            return start_date + timedelta(days=plans_duration["PREMIUM"])
+        else:
+            return None
+       
+        
+    def create_user(self, validated_data):
+        try:
+            
+            subscription_plan = validated_data.pop('subscription_plan')
+
+            if subscription_plan is None:
+                raise ValidationError("The 'subscription_plan' field is required.") # User: User object (31)
+
+            user = User.objects.create(subscription_plan=subscription_plan, **validated_data)
+            
+
+            # Create a new Subscription
+            start_date = timezone.now()
+            end_date = self.get_endtime(subscription_plan
+                                        , start_date)
+            if end_date is None:
+                raise ValidationError("Invalid subscription plan")
+
+            Subscription.objects.create(
+                user_id=user,
+                subscription_plan=subscription_plan,
+                start_date=start_date,
+                end_date=end_date    
+            )
+
+            logger.info(f"User created successfully with ID: {user.id}")
+
+            return user
+        except IntegrityError as e:
+            error_message = str(e)
+            logger.error(f"IntegrityError: {error_message}")
+            raise ValidationError(f"IntegrityError: {error_message}")
+        except Exception as e:
+            error_message = str(e)
+            logger.error(f"Unexpected error: {error_message}")
+            raise ValidationError(f"An error occurred: {error_message}")
+        
+        
+
+class SubscriptionService:
+    def get_subscription_data(self, request):
+        try:
+            subscriptions = Subscription.objects.all()
+            return subscriptions
+        except ObjectDoesNotExist as e:
+            logger.error(f"Object does not exist error: {e}")
+            return None
+        except MultipleObjectsReturned as e:
+            logger.error(f"Multiple objects returned error: {e}")
+            return None
+        except DatabaseError as e:
+            logger.error(f"Database error: {e}")
+            return None
+        except Exception as e:
+            logger.error(f"General error: {e}")
+            return None
+        
+    def get_subscriptionId_data(self, data):
+        subscription = Subscription.objects.get(id=data.get('id'))
+        return subscription
+    
+    
+    def get_subscriptionDatabase(self): # returns dataabse to check is the database exists
```

Comment:
what is the use of this

Reply:
Used to handle SubscriptionModel.DoesNotExists exeption handling